File: day3/gcd.py
def gcdIterative(n1, n2):
    min = n1 if (n1 < n2) else n2
    gcd = 1
    
    for i in range(2, min + 1):
        if(n1 % i == 0 and n2 % i == 0):
            gcd = i
            
    return gcd

def gcdRecursive(n1, n2):
    if(n1 % n2 == 0):
        return n2
    else:
        return gcdRecursive(n2, n1 % n2)
    
n1 = int(input("Enter 1st no.: "))
n2 = int(input("Enter 2nd no.: "))
gcd1 = gcdIterative(n1, n2)
gcd2 = gcdRecursive(n1, n2)

print(f"GCD of {n1} & {n2} by Iterative method = {gcd1}")
print(f"GCD of {n1} & {n2} by Recursive method = {gcd2}")

File: day5/gcd.r
gcd_iterative <- function(n1, n2) {
  if (n1 < n2) {
    min <- n1
  } else {
    min <- n2
  }

  gcd <- 1

  for (i in 2:min) {
    if (n1 %% i == 0 && n2 %% i == 0) {
      gcd <- i
    }
  }

  gcd
}

gcd_recursive <- function(n1, n2) {
  if (n1 %% n2 == 0) {
    n2
  } else {
    gcd_recursive(n2, n1 %% n2)
  }
}

n1 <- as.integer(readline("Enter 1st no.: "))
n2 <- as.integer(readline("Enter 2nd no.: "))

cat("GCD of", n1, "&", n2, "by Iterative method =", gcd_iterative(n1, n2), "\n")
cat("GCD of", n1, "&", n2, "by Recursive method =", gcd_recursive(n1, n2), "\n")
